Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -121,14 +178,53 @@ public class NFTStore: ObservableObject {
       let filters = self.filters
       let selectedAccounts = filters.accounts.filter(\.isSelected).map(\.model)
       let selectedNetworks = filters.networks.filter(\.isSelected).map(\.model)
-      let allVisibleUserAssets = assetManager.getAllVisibleAssetsInNetworkAssets(networks: selectedNetworks)
-      self.userVisibleNFTs = buildAssetViewModels(allVisibleUserAssets: allVisibleUserAssets)
-        .optionallyFilterUnownedNFTs(
-          isHidingUnownedNFTs: filters.isHidingUnownedNFTs,
-          selectedAccounts: selectedAccounts
-        )
       
-      let allTokens = allVisibleUserAssets.flatMap(\.tokens)
+      // user visible assets
+      let userVisibleAssets = assetManager.getAllUserAssetsInNetworkAssetsByVisibility(networks: selectedNetworks, visible: true)
+      // user hidden assets
+      let userHiddenAssets = assetManager.getAllUserAssetsInNetworkAssetsByVisibility(networks: selectedNetworks, visible: false)
+      // all spam NFTs marked by user
+      let allUserMarkedSpamNFTs = assetManager.getAllUserNFTs(networks: selectedNetworks, isSpam: true)
+      // all spam NFTs marked by SimpleHash
+      let simpleHashSpamNFTs = await walletService.simpleHashSpamNFTs(for: selectedAccounts, on: selectedNetworks)
```

Comment:
Is it possible to avoid this network request when using `Hide` or `Move to Spam`? There is a noticeable delay between selecting one of those options and the NFT being remove from visibility depending on network conditions that allows user to re-interact with the NFT before it's removed from screen:

https://github.com/brave/brave-ios/assets/5314553/e9f854d6-5c13-44f0-ba54-04dc7b4da8af